pkg/serviceprovider/github/downloadfilecapability.go

```diff
@@ -57,27 +60,37 @@ func NewDownloadFileCapability(httpClient *http.Client, ghClientBuilder githubCl
 		ghClientBuilder,
 		ghBaseUrl,
 		ghRepoRegexp,
+		lookup,
 	}, nil
 }
 
-func (d downloadFileCapability) DownloadFile(ctx context.Context, repoUrl, filepath, ref string, token *api.SPIAccessToken, maxFileSizeLimit int) (string, error) {
-	owner, repo, err := d.parseOwnerAndRepoFromUrl(ctx, repoUrl)
+func (d downloadFileCapability) DownloadFile(ctx context.Context, cl client.Client, request *api.SPIFileContentRequest, maxFileSizeLimit int) (string, error) {
+	owner, repo, err := d.parseOwnerAndRepoFromUrl(ctx, request.RepoUrl())
 	if err != nil {
 		return "", fmt.Errorf("could not parse repository name and owner from repoUrl: %w", err)
 	}
 	lg := log.FromContext(ctx)
-	ghClient, err := d.ghClientBuilder.createAuthenticatedGhClient(ctx, token)
+
+	credentials, err := d.lookup.LookupCredentials(ctx, cl, request)
```

Comment:
What about inverting this and providing the credentials to this method? That way we can do away with the heavy `lookup` in the downloadCapability.

Reply:
Yes, I think that is a cleaner solution, although we will need to extend the serviceProvider interface to LookupCredentials.